packages/firebase_database/ios/Classes/FirebaseDatabasePlugin.m

```diff
@@ -283,6 +283,16 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
             @"snapshot" : @{@"key" : snapshot.key ?: [NSNull null], @"value" : snapshot.value}
           });
         }];
+  } else if ([@"OnDisconnect#set" isEqualToString:call.method]) {
+    [getReference(database, call.arguments) onDisconnectSetValue:call.arguments[@"value"]
+                                                     andPriority:call.arguments[@"priority"]
```

Comment:
Since priority could not exist here we should check for it and use the appropriate `setValue` method.

Reply:
My experience in Obj-C is not that great. Will need help on this. 
I applied the same pattern as for `DatabaseReference#set` where priority is also optional.

Reply:
I see, @rmarau you are correct. In a future PR we should update both this and `DatabaseReference#set` to check for priority since having null priority removes previously set priorities.